db.go

```diff
@@ -660,6 +666,31 @@ func (a *dbAppender) Rollback() error {
 	return g.Wait()
 }
 
+// Delete implements deletion of metrics.
+func (db *DB) Delete(mint, maxt int64, ms ...labels.Matcher) error {
+	db.cmtx.Lock()
+	defer db.cmtx.Unlock()
+
+	db.headmtx.RLock()
```

Comment:
You sure: https://github.com/prometheus/tsdb/blob/master/querier.go#L62-L64 ? If yes, I will change there also.

But it also makes sense to use `headmtx` as blocks will change only when headblocks change?